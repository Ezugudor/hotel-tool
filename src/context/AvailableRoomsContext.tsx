import React, { ReactNode, useState } from "react";

export const AvailableRoomsContext = React.createContext<[AvailableRooms, (d: AvailableRooms) => void]>([{ economy: 0, premium: 0 }, (d) => { }])

export interface AvailableRoomsProviderProps {
    children: ReactNode
}

export interface AvailableRooms {
    economy: number,
    premium: number
}

export const AvailableRoomsProvider = (props: AvailableRoomsProviderProps) => {
    const [availableRooms, setAvailableRooms] = useState<AvailableRooms>({ economy: 0, premium: 0 });
    return <AvailableRoomsContext.Provider value={[availableRooms, setAvailableRooms]}>{props.children}</AvailableRoomsContext.Provider>
}