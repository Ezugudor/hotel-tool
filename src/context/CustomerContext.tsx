import React, { ReactNode } from "react";
import { CustomerBids } from "../mocks/CustomerBids";

export const CustomerContext = React.createContext<Array<number>>([])

export interface CustomerProviderProps {
    children: ReactNode
}

export const CustomerProvider = (props: CustomerProviderProps) => {
    return <CustomerContext.Provider value={CustomerBids}>{props.children}</CustomerContext.Provider>
}