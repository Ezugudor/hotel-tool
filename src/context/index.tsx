export * from "./CustomerContext"
export * from "./AvailableRoomsContext"