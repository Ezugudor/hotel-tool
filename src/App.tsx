import React from 'react';
import { Layout } from './components';
import { AvailableRoomsProvider, CustomerProvider } from './context';

const App: React.FunctionComponent = () => <div className="App">
  <CustomerProvider>
    <AvailableRoomsProvider>
      <Layout />
    </AvailableRoomsProvider>
  </CustomerProvider>
</div>;
export default App;
