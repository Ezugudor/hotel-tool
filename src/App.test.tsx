import React from 'react';
import App from './App';
import { shallow, ShallowWrapper } from 'enzyme';
import { Layout } from './components';
import { CustomerProvider, CustomerProviderProps } from './context';

let AppCont: ShallowWrapper;
let LayoutCont: ShallowWrapper;
let CustomerProviderCont: ShallowWrapper<CustomerProviderProps>;

beforeAll(() => {
  AppCont = shallow(<App />);
  CustomerProviderCont = AppCont.find(CustomerProvider);
  LayoutCont = CustomerProviderCont.find(Layout);
});


describe("App Component Test", () => {
  it('should match with existing snapshot', () => {
    expect(AppCont.html()).toMatchSnapshot();
  })

  it("should render single App component", () => {
    expect(AppCont).toHaveLength(1)
  })

  it("renders single CustomerProvider component", () => {
    expect(CustomerProviderCont).toHaveLength(1);
  })

  it("renders single Layout component within CustomerProvider component", () => {
    expect(LayoutCont).toHaveLength(1);
  })

})
