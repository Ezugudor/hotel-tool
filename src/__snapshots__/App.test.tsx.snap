// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test should match with existing snapshot 1`] = `"<div class=\\"App\\"><div><div><div><span class=\\"label\\">Total Requests Count: </span><span class=\\"value\\"> 10 </span></div><div><span class=\\"label\\">Total Requests Amount: </span><span class=\\"value\\"> 1243 </span></div><div><span class=\\"label\\">Premium Requests Count: </span><span class=\\"value\\"> 5 </span></div><div><span class=\\"label\\">Premium Requests Amount: </span><span class=\\"value\\"> 954 </span></div><div><span class=\\"label\\">Non Premium Requests Count: </span><span class=\\"value\\"> 5 </span></div><div><span class=\\"label\\">Non Premium Requests Amount: </span><span class=\\"value\\"> 289 </span></div><div> 0 fdfd  0</div></div><form><div><div><label for=\\"economyRoomField\\">Economy :</label><input type=\\"text\\" name=\\"economy\\" id=\\"economyRoomField\\" value=\\"\\"/></div><div><label for=\\"premiumRoomField\\">Premium :</label><input type=\\"text\\" name=\\"economy\\" id=\\"premiumRoomField\\" value=\\"\\"/></div><button type=\\"submit\\">Enter</button></div></form><div></div></div></div>"`;
